src/test/java/com/amazon/titan/testutils/TravisCiHeartbeat.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.titan.testutils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * This heartbeat timer will print to the console every x configured milliseconds. This is to prevent Travis CI from
+ * terminating the build if there is no console output in a 10min time frame. The heartbeat will only run for up to x
+ * configured intervals, this is needed for tests which are really in a hung state and Travis CI can then fail the build.
+ *
+ * The heartbeat is started and stopped before and after each test case. The current running unit test and the current
+ * execution time of the test is included in the heartbeat for better visibility in to any long running tests.
+ *
+ * @author Johan Jacobs
+ */
+public class TravisCiHeartbeat {
```

Comment:
This heartbeat is very useful and could be used in another CI platform. Perhaps note your classes generality and maybe even add a TODO to rename to CiHeartbeat. Dont need to make this change in this CR, just saying we should make note of it.